code/modules/antagonists/traitor/contractor/datums/syndicate_contract.dm

```diff
@@ -0,0 +1,518 @@
+#define CREDIT_REWARD_BASE 100
+#define CREDIT_REWARD_MULT_MIN 25
+#define CREDIT_REWARD_MULT_MAX 45
+#define DEFAULT_NAME "Unknown"
+#define DEFAULT_RANK "Unknown"
+#define EXTRACTION_PHASE_PREPARE 5 SECONDS
+#define EXTRACTION_PHASE_STEP_1 5 SECONDS
+#define COMPLETION_NOTIFY_DELAY 5 SECONDS
+#define RETURN_BRUISE_CHANCE 50
+#define RETURN_BRUISE_DAMAGE 20
+#define RETURN_SOUVENIR_CHANCE 10
+
+/**
+  * # Syndicate Contract
+  *
+  * Describes a contract that can be completed by a [/datum/antagonist/traitor/contractor].
+  */
+/datum/syndicate_contract
+	// Settings
+	/// Cooldown before making another extraction request in deciseconds.
+	var/extraction_cooldown = 10 MINUTES
+	/// How long an extraction portal remains before going away. Should be less than [/datum/syndicate_contract/var/extraction_cooldown].
+	var/portal_duration = 5 MINUTES
+	/// How long a target remains in the Syndicate jail.
+	var/prison_time = 4 MINUTES
+	/// List of items a target can get randomly after their return.
+	var/list/obj/item/souvenirs = list(
+		/obj/item/bedsheet/syndie,
+		/obj/item/clothing/under/syndicate/tacticool,
+		/obj/item/coin/antagtoken/syndicate,
+		/obj/item/reagent_containers/food/snacks/syndicake,
+		/obj/item/reagent_containers/food/snacks/tatortot,
+		/obj/item/storage/box/fakesyndiesuit,
+		/obj/item/storage/fancy/cigarettes/cigpack_syndicate,
+		/obj/item/toy/figure/syndie,
+		/obj/item/toy/nuke,
+		/obj/item/toy/plushie/nukeplushie,
+		/obj/item/toy/sword,
+		/obj/item/toy/syndicateballoon,
+		/obj/structure/sign/poster/contraband/syndicate_recruitment,
+	)
+	// Variables
+	/// The owning contractor hub.
+	var/datum/contractor_hub/owning_hub = null
+	/// The [/datum/objective/contract] associated to this contract.
+	var/datum/objective/contract/contract = null
+	/// Current contract status.
+	var/status = CONTRACT_STATUS_INVALID
+	/// Formatted station time at which the contract was completed, if applicable.
+	var/completed_time = ""
```

Comment:
Setting this one and the others like this doesn't make much sense.

Reply:
It eliminates the need to do the conversion in JavaScript, which is kind of a pain since `world.time` is in deciseconds and relative to the server start time.

Reply:
Same here. Meant setting them to an empty string and not just null (or undefined which is null). Unless it helps with the conversion to JS.